### PR TITLE
Telemetry to track the commands executed using ICommandManager

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #4926

### CODE REVIEW

File: src/client/common/application/commandManager.ts [hunk 1]
```diff
@@ -70,6 +72,7 @@ export class CommandManager implements ICommandManager {
         E extends keyof ICommandNameArgumentTypeMapping,
         U extends ICommandNameArgumentTypeMapping[E]
     >(command: E, ...rest: U): Thenable<T | undefined> {
+        sendTelemetryEvent(Telemetry.CommandExecuted, undefined, { command: command as string });
```
Comment (rchiodo) [inline src/client/common/application/commandManager.ts:75]: I was thinking we'd want this on registerCommand too. That way we'd track command palette calls as well.

Comment (rchiodo) [reply src/client/common/application/commandManager.ts:75]: Or maybe only on registerCommand (otherwise we might have dupes)

**Review by DavidKutu — APPROVED** [review]

**Review by rchiodo — APPROVED** [review]

File: src/client/common/application/commandManager.ts [hunk 2]
```diff
@@ -5,9 +5,12 @@
 
 import { injectable } from 'inversify';
 import { commands, Disposable, TextEditor, TextEditorEdit } from 'vscode';
+import { Telemetry } from '../../datascience/constants';
+import { sendTelemetryEvent } from '../../telemetry';
 import { ICommandNameArgumentTypeMapping } from './commands';
 import { ICommandManager } from './types';
 
+const commandsToIgnore = new Set<string>(['setContext']);
```
Comment (joyceerhl) [inline src/client/common/application/commandManager.ts:13]: Could we rename this variable to capture the fact that it's telemetry specific or add a comment to that effect?

**Review by joyceerhl — APPROVED** [review]

File: src/client/common/application/commandManager.ts (hunk 2 above)
Comment (DonJayamanne) [reply src/client/common/application/commandManager.ts:14]: Done

**Pull request merged**